p2panda-store/src/memory.rs

```diff
@@ -87,9 +93,7 @@ where
         log_id: &L,
     ) -> Result<bool, Self::Error> {
         if header.signature.is_none() {
-            return Err(anyhow!(
-                "store error: operation header must be signed prior to insertion"
-            ));
+            return Err(MemoryStoreError::MissingHeaderSignature);
         }
```

Comment:
This is caught in validation methods before, I believe this is not a concern of the storage itself

Reply:
Hmm...I hear what you're saying about the separation of concerns (validation and storage).

If we remove the check here then I'd expect we do the same for `insert_operation()` in `SqliteStore` (to keep the behaviour the same). If a header without a signature makes it into the `SqliteStore`, the corresponding `get_operation()` call is going to panic when the `from(row: OperationRow)` function tries to build the `Signature`. Then one would need to delete that operation.

If we're OK with that possibility then I'm happy to remove the `MissingHeaderSignature` variant from both error types.

Reply:
I'm also happy with panics occuring in the memory store if a header without a signature is inserted 👍

Reply:
Great, done! It turns out we get the validation for free anyways; SQLx returns an error on the insertion attempt because `signature` is defined as `NOT NULL`.